ctd/read.py

```diff
@@ -61,6 +61,9 @@ def _open_compressed(fname):
 
 def _read_file(fname):
     """Read file contents."""
+    if isinstance(fname, StringIO):
+        return fname
```

Comment:
It is probably safer to add the fname.seek(0) and then `text = fname.copy()` (or something similar.)
We should also change the docstring to say "read from file or StringIO obj.

Reply:
How about:

```python
    """Read file contents, or read from StringIO object."""
    if isinstance(fname, StringIO):
        fname.seek(0)
        text = fname.read()
        return StringIO(text)
```

Reply:
Looks good. Yeah, sorry for not checking I prototyped from my phone :-)
That looks good, not memory efficient but safe. We don't really need to be memory efficient b/c the files we are dealing with are never more than a few megabytes.